src/pybind/mgr/cephadm/module.py

```diff
@@ -3080,6 +3086,21 @@ def _get_prometheus_credentials(self) -> Tuple[str, str]:
             self.set_store(PrometheusService.PASS_CFG_KEY, password)
         return (user, password)
 
+    @handle_orch_error
+    def generate_certificates(self, module_name: str) -> Optional[Dict[str, str]]:
```

Comment:
As far as I can tell, this function doesn't actually use the module_name for anything in terms of generating the certificate. I don't see why not just allow this to be a generic function for generating certs rather than requiring I pass `prometheus` or `dashboard` in. We could have other use cases later and the restriction seems unnecessary to me.

Reply:
Well, for security reasons I was thinking on generating certs based on where the daemons are running .. but this is something that we can add later. The dashboard module does have a small difference .. but that could go as an extra parameter making the code generic.

```
        if module_name == 'dashboard':
            host_fqdns.append('dashboard_servers')
```